### PR TITLE
add sdk base code

### PR DESCRIPTION
**Pull request by peacepassion**
Ref #83 

Still has task todo: 
1. request params definition;  
2. https support,  
3. united test with backend;  
4. bs patch support;

### CODE REVIEW

**Comment by peacepassion** [comment]
see #90 directly

File: amigo-sdk/src/main/java/me/ele/amigo/sdk/http/Http.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+                        return;
+                    }
+
+                    long contentLength = connection.getContentLength();
+                    InputStream is = connection.getInputStream();
+
+                    final ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+                    byte[] buffer = new byte[4 * 1024];
+                    if (is == null) {
```
Comment (FtheGFW) [inline amigo-sdk/src/main/java/me/ele/amigo/sdk/http/Http.java]: pull this code block up to line 80 would be nice

File: amigo-sdk/src/main/java/me/ele/amigo/sdk/http/Http.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                                    callback.onSucc(new Response(responseCode, os.toByteArray()));
+                                    callback.onComplete();
+                                }
+                            });
+                        } else {
+                            handler.post(new Runnable() {
+                                @Override
+                                public void run() {
+                                    try {
+                                        callback.onFail(new Error(responseCode, null, new String(os.toByteArray(), "UTF-8")));
```
Comment (FtheGFW) [inline amigo-sdk/src/main/java/me/ele/amigo/sdk/http/Http.java]: why not use ComonUtil.byteArray2String here?

File: amigo-sdk/src/main/java/me/ele/amigo/sdk/utils/MD5.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            Log.e(TAG, "Exception while getting FileInputStream", e);
+            return null;
+        }
+
+        byte[] buffer = new byte[8192];
+        int read;
+        try {
+            while ((read = is.read(buffer)) > 0) {
+                digest.update(buffer, 0, read);
+            }
```
Comment (FtheGFW) [inline amigo-sdk/src/main/java/me/ele/amigo/sdk/utils/MD5.java]: I would prefer extract this  code into a separate utility method, named like #readRawBytes(File file), and just let calculateMD5() taking a byte[] as the input param, in that case, the previous #verifyPatchApk() can take advantage of the http response body instead.

Comment (FtheGFW) [inline amigo-sdk/src/main/java/me/ele/amigo/sdk/utils/MD5.java]: I would prefer extract this  code into a separate utility method, named like #readRawBytes(File file), and just let calculateMD5() taking a byte[] as the input param, in that case, the previous #verifyPatchApk() can take advantage of the http response body instead.  what do you think ?

**Pull request closed without merging**